src/charm.py

```diff
@@ -198,6 +214,36 @@ def _on_leader_elected(self, _):
                 password = generate_random_password(PASSWORD_LENGTH)
                 peer_data[required_password] = password
 
+        # If this node was elected a leader due to a prior leader unit being down scaled
+        if self._is_peer_data_set and self._is_cluster_initialized:
+            self.unit.status = MaintenanceStatus("Removing unreachable instances")
+
+            # Remove unreachable instances from the cluster
+            self._mysql.remove_instances_not_online()
+
+            # Since we do not have name of the prior leader unit, use planned units
+            # to remove units with index >= planned units from the allowlist
+            planned_units = self.app.planned_units()
+            allowlist = set(self._peers.data[self.app].get("allowlist").split(","))
```

Comment:
sorry noob to this lingo, whats an "allow list"

Reply:
An [allowlist](https://dev.mysql.com/doc/refman/8.0/en/group-replication-options.html#sysvar_group_replication_ip_allowlist) is a list of hostnames/IPs that are permitted to be part of the group (cluster) in MySQL